File: js/Linked List Node Removal Solution.js
var removeNodes = function(head) {
    const stack = [];
    while (head) {
        while (stack.length && stack[stack.length - 1].val < head.val) {
            stack.pop();
        }
        stack.push(head);
        head = head.next;
    }
    let nxt = null;
    while (stack.length) {
        const cur = stack.pop();
        cur.next = nxt;
        nxt = cur;
    }
    return nxt;
};

File: python/Linked List Node Removal Solution.py
class Solution:
    def removeNodes(self, head: ListNode) -> ListNode:
        stack = []
        while head:
            while stack and stack[-1].val < head.val:
                stack.pop()
            stack.append(head)
            head = head.next
        nxt = None
        while stack:
            cur = stack.pop()
            cur.next = nxt
            nxt = cur
        return nxt